src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,32 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASS_LENGTH = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        StorageDao storageDao = new StorageDaoImpl();
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RuntimeException();
+        }
+        if (storageDao.get(user.getLogin()) == null
+                || storageDao.get(user.getLogin()).equals(user.getLogin())) {
```

Comment:
```suggestion
        if (storageDao.get(user.getLogin()) != null) {
            throw new RuntimeException("This login is already exist");
        }
```
таким способом ми позбудемось вкладеності коду